### PR TITLE
Wrap message at 67 characters with mkchlog

### PR DESCRIPTION
**Pull request by mackdk**
## What does this PR change?

This PR makes `mkchlog` wrap the message at 67 characters like specified [in the wiki](https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs). This works only for the text specified on the command line. When editing the file with the editor, no changes are made to the user input.

An additional option `-n, --no-wrap` is added to allow avoiding the automatic text wrapping if needed.

 For example:

```
mkchlog "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus ultricies ac metus a varius. Phasellus quis nisi ac libero maximus lobortis accumsan."
```

Will produce the following changelog:
```
- Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus 
  ultricies ac metus a varius. Phasellus quis nisi ac libero 
  maximus lobortis accumsan.
```

while:
```
mkchlog -n "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus ultricies ac metus a varius. Phasellus quis nisi ac libero maximus lobortis accumsan."
```

Will produce:
```
- Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus ultricies ac metus a varius. Phasellus quis nisi ac libero maximus lobortis accumsan.
```

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by cbbayburt — APPROVED** [review]
Great addition! Thanks, @mackdk 

+1 for making this the default as well. However, I'm not really sure if this 67-char limit is still exactly relevant (otherwise we might want to pick a bigger limit).

Let's wait for @deneb-alpha's review too.

**Comment by cbbayburt** [comment]
> Great addition! Thanks, @mackdk 
> 
> 
> 
> I would vote for making this the default as well (having a `--no-wrap` argument instead).
> 

This actually depends a bit on how the entries are treated during tagging. I guess they are wrapped then anyway, so in that case we don't need to make it the default. For us devs it'll be just cosmetics.

**Comment by deneb-alpha** [comment]
> Great addition! Thanks, @mackdk
> 
> I would vote for making this the default as well (having a `--no-wrap` argument instead). However, I'm not really sure if this 67-char limit is still exactly relevant (otherwise we might want to pick a bigger limit).
> 
> Let's wait for @deneb-alpha's review too.

I'm for following the guidelines, I like to have the default wrapping

File: rel-eng/bin/mkchlog [hunk 1]
```diff
@@ -20,6 +20,7 @@ The default editor can be specified by setting the EDITOR environment variable.
   -f, --feature         set the feature name to use as a filename part
   -u, --username        set the username to use as a filename part
   -r, --remove          remove existing changelog file
+  -w, --auto-wrap       wraps automatically the message at 67 characters
```
Comment (deneb-alpha) [inline rel-eng/bin/mkchlog:23]: as @cbbayburt also suggested, let's go  with `--no-wrap` as  default

File: rel-eng/bin/mkchlog [hunk 2]
```diff
@@ -46,6 +47,10 @@ while [[ $# -gt 0 ]]; do
             REMOVE=1
             shift
             ;;
+        -w|--auto-wrap)
+            WRAP=true
+            shift
+            ;;
```
Comment (deneb-alpha) [inline rel-eng/bin/mkchlog:53]: and with the `--no-wrap` as default, this part  should also be adapted

File: rel-eng/bin/mkchlog [hunk 3]
```diff
@@ -140,7 +145,11 @@ if ! [ -z $REMOVE ]; then
 fi
 
 # Add the new entry
-echo "- $1" > $CHFILE.new
+if [ "$WRAP" = true ]; then
+    echo "$1" | fold -s -w 65 | sed '1s/^.*$/- &/;2,$s/^.*$/  &/' > $CHFILE.new
+else
+    echo "- $1" > $CHFILE.new
+fi
```
Comment (deneb-alpha) [inline rel-eng/bin/mkchlog:152]: the same applies here for  `--no-wrap`

**Comment by mackdk** [comment]
I made the wrapping the default behaviour and added a `-n|--no-wrap` option to avoid it. The PR title and description has been updated to reflect that.

**Review by deneb-alpha — APPROVED** [review]
thanks a lot for  this really nice  feature. it will help in preparing the patchinfo speeding up my  work!

**Pull request merged**